command/pr.go

```diff
@@ -470,36 +474,146 @@ func prMerge(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	rebase, err := cmd.Flags().GetBool("rebase")
+	var mergeMethod api.PullRequestMergeMethod
+	deleteBranch, err := cmd.Flags().GetBool("delete-branch")
 	if err != nil {
-		return err
+		return nil
 	}
-	squash, err := cmd.Flags().GetBool("squash")
-	if err != nil {
-		return err
+
+	// Ensure only one merge method is specified
+	found := 0
+	isInteractive := false
+	if cmd.Flags().Changed("merge") {
+		found++
+	}
+	if cmd.Flags().Changed("rebase") {
+		found++
+	}
+	if cmd.Flags().Changed("squash") {
+		found++
+	}
+	if found == 0 {
+		isInteractive = true
+	} else if found > 1 {
+		return errors.New("expected exactly one of --merge, --rebase, or --squash")
+	}
+
+	if isInteractive {
+		mergeMethod, deleteBranch, err = prInteractiveMerge()
+		if err != nil {
+			return nil
+		}
+
+	} else {
+		rebase, err := cmd.Flags().GetBool("rebase")
+		if err != nil {
+			return err
+		}
+		squash, err := cmd.Flags().GetBool("squash")
+		if err != nil {
+			return err
+		}
+
+		if rebase {
+			mergeMethod = api.PullRequestMergeMethodRebase
+		} else if squash {
+			mergeMethod = api.PullRequestMergeMethodSquash
+		} else {
+			mergeMethod = api.PullRequestMergeMethodMerge
+		}
 	}
 
-	var output string
-	if rebase {
-		output = fmt.Sprintf("%s Rebased and merged pull request #%d\n", utils.Green("✔"), pr.Number)
+	var action string
+	if mergeMethod == api.PullRequestMergeMethodRebase {
+		action = "Rebased and merged"
 		err = api.PullRequestMerge(apiClient, baseRepo, pr, api.PullRequestMergeMethodRebase)
-	} else if squash {
-		output = fmt.Sprintf("%s Squashed and merged pull request #%d\n", utils.Green("✔"), pr.Number)
+	} else if mergeMethod == api.PullRequestMergeMethodSquash {
+		action = "Squashed and merged"
 		err = api.PullRequestMerge(apiClient, baseRepo, pr, api.PullRequestMergeMethodSquash)
-	} else {
-		output = fmt.Sprintf("%s Merged pull request #%d\n", utils.Green("✔"), pr.Number)
+	} else if mergeMethod == api.PullRequestMergeMethodMerge {
+		action = "Merged"
 		err = api.PullRequestMerge(apiClient, baseRepo, pr, api.PullRequestMergeMethodMerge)
+	} else {
+		fmt.Fprintf(os.Stderr, "Unknown merge method used\n")
+		os.Exit(1)
 	}
 
 	if err != nil {
 		return fmt.Errorf("API call failed: %w", err)
 	}
 
-	fmt.Fprint(colorableOut(cmd), output)
+	fmt.Fprintf(colorableOut(cmd), "%s %s pull request #%d\n", utils.Magenta("✔"), action, pr.Number)
+
+	if deleteBranch {
+		branch, err := prDeleteCurrentBranch(baseRepo.(*api.Repository))
```

Comment:
`baseRepo` is of type `ghrepo.Interface` and its value will often be `api.Repository`, but sometimes not. In the latter cases, this line will panic.

Reply:
I'm trying to figure out a clean way around this because `determineBaseRepo` will return an api.Repository unless the `repo` flag is set, and the merge command won't work well with the repo flag.